plugin/src/main/java/org/opensearch/ml/action/model_group/TransportUpdateModelGroupAction.java

```diff
@@ -134,13 +139,32 @@ private void updateModelGroup(
         if (Boolean.TRUE.equals(updateModelGroupInput.getIsAddAllBackendRoles())) {
             source.put(MLModelGroup.BACKEND_ROLES_FIELD, user.getBackendRoles());
         }
-        if (StringUtils.isNotBlank(updateModelGroupInput.getName())) {
-            source.put(MLModelGroup.MODEL_GROUP_NAME_FIELD, updateModelGroupInput.getName());
-        }
         if (StringUtils.isNotBlank(updateModelGroupInput.getDescription())) {
             source.put(MLModelGroup.DESCRIPTION_FIELD, updateModelGroupInput.getDescription());
         }
+        if (StringUtils.isNotBlank(updateModelGroupInput.getName()) && !updateModelGroupInput.getName().equals(modelGroupName)) {
+            mlModelGroupManager.validateUniqueModelGroupName(updateModelGroupInput.getName(), ActionListener.wrap(modelGroups -> {
```

Comment:
what happens if model group table is not created yet? won't we face index not found exception?

Reply:
Hmm yeah, that is a possibility actually. But for user to send update request, they need model group ID which would have created the model group index already